cellxgene_resources/compare_revision.ipynb

```diff
@@ -80,118 +36,92 @@
-   "execution_count": null,
-   "id": "05d23867",
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "should_differ = [\n",
+    "should_differ_dataset = [\n",
     "    'dataset_version_id','explorer_url','assets','revised_at','citation','processing_status'\n",
     "]\n",
     "\n",
     "ont_fields = [\n",
     "    'assay','cell_type','development_stage','disease',\n",
-    "    'self_reported_ethnicity','sex','tissue'\n",
-    "]\n",
-    "\n",
-    "pub_datasets = {d['dataset_id']: d for d in collection['datasets']}\n",
-    "rev_datasets = {d['dataset_id']: d for d in rev['datasets']}\n",
-    "\n",
-    "comp = {}\n",
-    "for ds_id,v in rev_datasets.items():\n",
-    "    comp[ds_id] = {'title': v['title']}\n",
-    "    if ds_id not in pub_datasets.keys():\n",
-    "        comp[ds_id]['in published'] = 'not present'\n",
-    "    else:\n",
-    "        for prop,rev_val in v.items():\n",
-    "            if prop not in should_differ:\n",
-    "                pub_val = pub_datasets[ds_id].get(prop)\n",
-    "                if prop in ont_fields:\n",
-    "                    rev_val = [t['label'] for t in rev_val]\n",
-    "                    pub_val = [t['label'] for t in pub_val]\n",
-    "                if isinstance(rev_val, list) and prop != 'assets':\n",
-    "                    rev_val.sort()\n",
-    "                    pub_val.sort()\n",
-    "                if pub_val != rev_val:\n",
-    "                    if prop == 'mean_genes_per_cell' and round(rev_val, 5) == round(pub_val, 5):\n",
-    "                        continue\n",
-    "                    comp[ds_id][prop + '_REV'] = rev_val\n",
-    "                    comp[ds_id][prop + '_PUB'] = pub_val\n",
-    "\n",
-    "comp_df = pd.DataFrame(comp).transpose()\n",
-    "comp_df = comp_df.dropna(subset=[c for c in comp_df.columns if c != 'title'], how='all')\n",
-    "comp_df.fillna('')"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "id": "7f03a4c4-3036-4f79-88c0-26d8125de5c4",
-   "metadata": {},
-   "source": [
-    "**Identify properties to investigate differences between long lists**"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "5ef9de56-7568-4c85-8f3a-7c060bdb640b",
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "fields = [\n",
-    "    'tissue'\n",
+    "    'self_reported_ethnicity','sex','tissue','organism'\n",
     "]"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "b2d7882d-4bbf-4846-a401-229b7479849e",
+   "id": "acc7c9bf",
    "metadata": {},
    "outputs": [],
    "source": [
-    "for f in fields:\n",
+    "for k,v in rev.items():\n",
+    "    if k not in collection.keys():\n",
+    "        if k not in should_be_absent:\n",
+    "            print('not present: ' + k)\n",
+    "    elif collection.get(k) != v and k not in should_differ_collection:\n",
+    "        if k == 'datasets':\n",
+    "            diff_props = []\n",
```

Comment:
And related to the print out of list differences, I think making `diff_props` a set instead of a list should capture all the differences without repetition

Reply:
fixed via e7323cbc3a73ce66a41960438639016ebdaced2f